.github/workflows/release.yml

```diff
@@ -75,15 +80,18 @@ jobs:
         with:
           ref: ${{ inputs.versionNumber }}
           fetch-depth: 0
+
       - name: Generate Release Notes
         run: |
           cd .changes
           sed -e "1{/# /d;}" -e "2{/^$/d;}" ${{ needs.changelog-version.outputs.version }}.md > release-notes.txt
+
       - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
```

Comment:
Fun note, the docs for this action doesn't specify what permission is needed, but general consensus seems to be that `content: read` is enough

https://github.com/actions/upload-artifact/issues/197

Reply:
Fascinating 😄 We could add a job-level permissions block if this turns out to not be true.